explorer/interpreter/interpreter.cpp

```diff
@@ -2210,6 +2210,15 @@ auto Interpreter::StepDeclaration() -> ErrorOr<Success> {
     case DeclarationKind::VariableDeclaration: {
       const auto& var_decl = cast<VariableDeclaration>(decl);
       if (var_decl.has_initializer()) {
+        auto initializer = &var_decl.binding().static_type();
+        if (const auto* dest_class = dyn_cast<NominalClassType>(initializer)) {
```

Comment:
This check should be performed by the type-checker, not by the interpreter. A good place would be `TypeChecker::DeclareDeclaration`, where we currently call `ExpectCompleteType`.

Reply:
Done